### ISSUE TITLE
Docker Desktop stuck on "Starting the Docker Engine"

### ISSUE THREAD

**Issue by farfnd** [docker/for-win#13468]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 452FB130-BF76-4086-95E7-39C6C907E98E/20230506090110

### Actual behavior

![image](https://user-images.githubusercontent.com/70105993/236660284-c618109a-6bad-4750-bcc0-8b620169a042.png)

### Information

  - Windows Version: Windows 11 Home Single Language, Version 21H2, Build 22000.1817
  - Docker Desktop Version: 4.19.0
  - WSL2 or Hyper-V backend? WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: no

  - Output of `wsl --version`
    ```
    WSL version: 1.2.5.0
    Kernel version: 5.15.90.1
    WSLg version: 1.0.51
    MSRDC version: 1.2.3770
    Direct3D version: 1.608.2-61064218
    DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
    Windows version: 10.0.22000.1817
    ```
  
  - Output of `wsl --status`
    ```
    Default Distribution: Ubuntu
    Default Version: 2
    ```
  
  - Output of `wsl -l -v --all`
    ```
      NAME                   STATE           VERSION
    * Ubuntu                 Stopped         2
      docker-desktop         Stopped         2
      docker-desktop-data    Stopped         2
    ```

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`
```
[2023-05-06T09:22:57.306093900Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[SKIP] DD0002: does the bootloader have virtualization enabled?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[SKIP] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[SKIP] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[FAIL] DD0011: are the LinuxKit services running? failed to ping VM diagnosticsd with error: Get "http://ipc/ping": open \\.\pipe\dockerDiagnosticd: The system cannot find the file specified.
[2023-05-06T09:23:03.360531600Z][com.docker.diagnose.exe][I] ipc.NewClient: d328f94a-diagnose -> \\.\pipe\dockerDiagnosticd diagnosticsd
[2023-05-06T09:23:03.361033000Z][com.docker.diagnose.exe][I] (77728c9f) d328f94a-diagnose C->S diagnosticsd GET /ping
[2023-05-06T09:23:03.361033000Z][com.docker.diagnose.exe][W] (77728c9f) d328f94a-diagnose C<-S NoResponse GET /ping (0s): Get "http://ipc/ping": open \\.\pipe\dockerDiagnosticd: The system cannot find the file specified.

[FAIL] DD0004: is the Docker engine running? Get "http://ipc/docker": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-06T09:23:03.361530900Z][com.docker.diagnose.exe][I] ipc.NewClient: e51dd2ac-com.docker.diagnose -> \\.\pipe\dockerLifecycleServer VMDockerdAPI
[2023-05-06T09:23:03.362031400Z][com.docker.diagnose.exe][I] (a62ef4fa) e51dd2ac-com.docker.diagnose C->S VMDockerdAPI GET /docker
[2023-05-06T09:23:03.362031400Z][com.docker.diagnose.exe][W] (a62ef4fa) e51dd2ac-com.docker.diagnose C<-S NoResponse GET /docker (0s): Get "http://ipc/docker": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-06T09:23:03.362031400Z][com.docker.diagnose.exe][I] (a62ef4fa-1) e51dd2ac-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-06T09:23:03.362532300Z][com.docker.diagnose.exe][W] (a62ef4fa-1) e51dd2ac-com.docker.diagnose C<-S NoResponse GET /ping (500.9µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-06T09:23:04.367747200Z][com.docker.diagnose.exe][I] (a62ef4fa-2) e51dd2ac-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-06T09:23:04.368095600Z][com.docker.diagnose.exe][W] (a62ef4fa-2) e51dd2ac-com.docker.diagnose C<-S NoResponse GET /ping (348.4µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-06T09:23:05.373692300Z][com.docker.diagnose.exe][I] (a62ef4fa-3) e51dd2ac-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-06T09:23:05.374025000Z][com.docker.diagnose.exe][W] (a62ef4fa-3) e51dd2ac-com.docker.diagnose C<-S NoResponse GET /ping (332.7µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-06T09:23:06.387502100Z][com.docker.diagnose.exe][I] (a62ef4fa-4) e51dd2ac-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-06T09:23:06.387502100Z][com.docker.diagnose.exe][W] (a62ef4fa-4) e51dd2ac-com.docker.diagnose C<-S NoResponse GET /ping (296.2µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-06T09:23:07.400977900Z][com.docker.diagnose.exe][I] (a62ef4fa-5) e51dd2ac-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-06T09:23:07.401246400Z][com.docker.diagnose.exe][W] (a62ef4fa-5) e51dd2ac-com.docker.diagnose C<-S NoResponse GET /ping (268.5µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-06T09:23:08.416627100Z][com.docker.diagnose.exe][I] (a62ef4fa-6) e51dd2ac-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-06T09:23:08.416627100Z][com.docker.diagnose.exe][W] (a62ef4fa-6) e51dd2ac-com.docker.diagnose C<-S NoResponse GET /ping (0s): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-06T09:23:09.418405900Z][com.docker.diagnose.exe][I] (a62ef4fa-7) e51dd2ac-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-06T09:23:09.418405900Z][com.docker.diagnose.exe][W] (a62ef4fa-7) e51dd2ac-com.docker.diagnose C<-S NoResponse GET /ping (0s): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-06T09:23:10.420039000Z][com.docker.diagnose.exe][I] (a62ef4fa-8) e51dd2ac-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-06T09:23:10.420039000Z][com.docker.diagnose.exe][W] (a62ef4fa-8) e51dd2ac-com.docker.diagnose C<-S NoResponse GET /ping (0s): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.

[PASS] DD0015: are the binary symlinks installed?
[FAIL] DD0031: does the Docker API work? error during connect: in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine_linux/v1.24/containers/json": open //./pipe/docker_engine_linux: The system cannot find the file specified.
[PASS] DD0013: is the $PATH ok?
error during connect: in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/containers/json": open //./pipe/docker_engine: The system cannot find the file specified.
[FAIL] DD0003: is the Docker CLI working? exit status 1
[PASS] DD0005: is the user in the docker-users group?
[FAIL] DD0038: is the connection to Docker working? HTTP GET https://login.docker.com: Get "https://login.docker.com": proxyconnect tcp: open \\.\pipe\dockerHTTPProxy: The system cannot find the file specified.
[FAIL] DD0014: are the backend processes running? 2 errors occurred:
        * com.docker.vpnkit.exe is not running
        * com.docker.proxy.exe is not running


[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[FAIL] DD0009: is the vpnkit API responding? open \\.\pipe\dockerVpnKitDiagnostics: The system cannot find the file specified.
[FAIL] DD0010: is the Docker API proxy responding? failed to ping Docker proxy API with error: Get "http://ipc/desktop-diagnostics/ping": open \\.\pipe\dockerDesktopLinuxEngine: The system cannot find the file specified.
[2023-05-06T09:23:10.749794700Z][com.docker.diagnose.exe][I] ipc.NewClient: 19279b87-diagnose -> \\.\pipe\dockerDesktopLinuxEngine Proxy
[2023-05-06T09:23:10.749794700Z][com.docker.diagnose.exe][I] (f77a5003) 19279b87-diagnose C->S Proxy GET /desktop-diagnostics/ping
[2023-05-06T09:23:10.750409900Z][com.docker.diagnose.exe][W] (f77a5003) 19279b87-diagnose C<-S NoResponse GET /desktop-diagnostics/ping (615.2µs): Get "http://ipc/desktop-diagnostics/ping": open \\.\pipe\dockerDesktopLinuxEngine: The system cannot find the file specified.
[2023-05-06T09:23:10.750409900Z][com.docker.diagnose.exe][I] (f77a5003-1) 19279b87-diagnose C->S Proxy GET /ping
[2023-05-06T09:23:10.750409900Z][com.docker.diagnose.exe][W] (f77a5003-1) 19279b87-diagnose C<-S NoResponse GET /ping (0s): Get "http://ipc/ping": open \\.\pipe\dockerDesktopLinuxEngine: The system cannot find the file specified.
[2023-05-06T09:23:11.757910400Z][com.docker.diagnose.exe][I] (f77a5003-2) 19279b87-diagnose C->S Proxy GET /ping
[2023-05-06T09:23:11.757910400Z][com.docker.diagnose.exe][W] (f77a5003-2) 19279b87-diagnose C<-S NoResponse GET /ping (0s): Get "http://ipc/ping": open \\.\pipe\dockerDesktopLinuxEngine: The system cannot find the file specified.
[2023-05-06T09:23:12.772116600Z][com.docker.diagnose.exe][I] (f77a5003-3) 19279b87-diagnose C->S Proxy GET /ping
[2023-05-06T09:23:12.772977100Z][com.docker.diagnose.exe][W] (f77a5003-3) 19279b87-diagnose C<-S NoResponse GET /ping (860.5µs): Get "http://ipc/ping": open \\.\pipe\dockerDesktopLinuxEngine: The system cannot find the file specified.
[2023-05-06T09:23:13.779672100Z][com.docker.diagnose.exe][I] (f77a5003-4) 19279b87-diagnose C->S Proxy GET /ping
[2023-05-06T09:23:13.780495500Z][com.docker.diagnose.exe][W] (f77a5003-4) 19279b87-diagnose C<-S NoResponse GET /ping (823.4µs): Get "http://ipc/ping": open \\.\pipe\dockerDesktopLinuxEngine: The system cannot find the file specified.
[2023-05-06T09:23:14.789589400Z][com.docker.diagnose.exe][I] (f77a5003-5) 19279b87-diagnose C->S Proxy GET /ping
[2023-05-06T09:23:14.794156700Z][com.docker.diagnose.exe][W] (f77a5003-5) 19279b87-diagnose C<-S NoResponse GET /ping (4.5673ms): Get "http://ipc/ping": open \\.\pipe\dockerDesktopLinuxEngine: The system cannot find the file specified.
[2023-05-06T09:23:15.808770000Z][com.docker.diagnose.exe][I] (f77a5003-6) 19279b87-diagnose C->S Proxy GET /ping
[2023-05-06T09:23:15.808770000Z][com.docker.diagnose.exe][W] (f77a5003-6) 19279b87-diagnose C<-S NoResponse GET /ping (0s): Get "http://ipc/ping": open \\.\pipe\dockerDesktopLinuxEngine: The system cannot find the file specified.
[2023-05-06T09:23:16.822528100Z][com.docker.diagnose.exe][I] (f77a5003-7) 19279b87-diagnose C->S Proxy GET /ping
[2023-05-06T09:23:16.822814800Z][com.docker.diagnose.exe][W] (f77a5003-7) 19279b87-diagnose C<-S NoResponse GET /ping (286.7µs): Get "http://ipc/ping": open \\.\pipe\dockerDesktopLinuxEngine: The system cannot find the file specified.
[2023-05-06T09:23:17.828957800Z][com.docker.diagnose.exe][I] (f77a5003-8) 19279b87-diagnose C->S Proxy GET /ping
[2023-05-06T09:23:17.830615000Z][com.docker.diagnose.exe][W] (f77a5003-8) 19279b87-diagnose C<-S NoResponse GET /ping (1.6572ms): Get "http://ipc/ping": open \\.\pipe\dockerDesktopLinuxEngine: The system cannot find the file specified.

[SKIP] DD0006: is the Docker Desktop Service responding?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[WARN] DD0011: are the LinuxKit services running? failed to ping VM diagnosticsd with error: Get "http://ipc/ping": open \\.\pipe\dockerDiagnosticd: The system cannot find the file specified.
[WARN] DD0004: is the Docker engine running? Get "http://ipc/docker": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[PASS] DD0015: are the binary symlinks installed?
[WARN] DD0031: does the Docker API work? error during connect: in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine_linux/v1.24/containers/json": open //./pipe/docker_engine_linux: The system cannot find the file specified.
[WARN] DD0032: do Docker networks overlap with host IPs? error during connect: in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine_linux/v1.24/networks": open //./pipe/docker_engine_linux: The system cannot find the file specified.

Please note the following 4 warnings:

1 : The check: are the LinuxKit services running?
    Produced the following warning: failed to ping VM diagnosticsd with error: Get "http://ipc/ping": open \\.\pipe\dockerDiagnosticd: The system cannot find the file specified.

The Docker engine runs inside a Linux VM as a service. Therefore the services must have started.

2 : The check: is the Docker engine running?
    Produced the following warning: Get "http://ipc/docker": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.

The Docker engine manages all containers and images on the host. Check the dockerd.log to see why it failed to start.

3 : The check: does the Docker API work?
    Produced the following warning: error during connect: in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine_linux/v1.24/containers/json": open //./pipe/docker_engine_linux: The system cannot find the file specified.

If the Docker API is not available from the host then Docker Desktop will not work correctly.

4 : The check: do Docker networks overlap with host IPs?
    Produced the following warning: error during connect: in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine_linux/v1.24/networks": open //./pipe/docker_engine_linux: The system cannot find the file specified.

If the subnet used by a Docker network overlaps with an IP used by the host, then containers
won't be able to contact the overlapping IP addresses.

Try configuring the IP address range used by networks: in your docker-compose.yml.
See https://docs.docker.com/compose/compose-file/compose-file-v2/#ipv4_address-ipv6_address


Please investigate the following 2 issues:

1 : The test: are the LinuxKit services running?
    Failed with: failed to ping VM diagnosticsd with error: Get "http://ipc/ping": open \\.\pipe\dockerDiagnosticd: The system cannot find the file specified.

The Docker engine runs inside a Linux VM as a service. Therefore the services must have started.

2 : The test: are the backend processes running?
    Failed with: 2 errors occurred:
        * com.docker.vpnkit.exe is not running
        * com.docker.proxy.exe is not running


Not all of the backend processes are running.
```

### Steps to reproduce the behavior

  1. ...
  2. ...

**Comment by shivamnegi2424** [comment]
I'm facing the exact same issue

**Comment by farfnd** [comment]
Solved by clearing these directories and reinstalling:
- `%APPDATA%\Docker`
- `%USERPROFILE%\.docker`
- `%USERPROFILE%\AppData\Local\Docker`

**Comment by jeanineharb** [comment]
Thank you, solved my issue!

**Comment by samueleiche** [comment]
I sometimes get this issue after an update to docker and I usually solve this by terminating any docker processes and restarting Windows. Hope this can help anyone reading for solutions.

**Comment by jonahhawk** [comment]
@samueleiche This is the way. I really did not want to nuke Docker from orbit and start over. Thank you!

**Comment by mahdirezaei-dev** [comment]
> Solved by clearing these directories and reinstalling:
> 
> * `%APPDATA%\Docker`
> * `%USERPROFILE%\.docker`
> * `%USERPROFILE%\AppData\Local\Docker`

It was helpful for me ✌

**Comment by Console-Coder** [comment]
guys, if you have tiny wall firewall it might cause issues when starting docker, to fix this, make sure you allow docker backend process using show connections window show blocked connections and right click and unblock the trick is to click on the notification and customize the rule to "unrestricted" instead of "unrestricted TCP and UDP" this just worked for me just tried it so I thought to write it here in case someone else uses docker and has tinywall firewall installed

**Comment by patric0889** [comment]
> guys, if you have tiny wall firewall it might cause issues when starting docker, to fix this, make sure you allow docker backend process using show connections window show blocked connections and right click and unblock the trick is to click on the notification and customize the rule to "unrestricted" instead of "unrestricted TCP and UDP" this just worked for me just tried it so I thought to write it here in case someone else uses docker and has tinywall firewall installed

Thx, that worked for me!

**Comment by sina-karimi-93** [comment]
2 days ago, after updating windows 11 and restart it, the docker stuck on starting...
this is the issue id: B8C30C96-A8AE-4244-8558-194D4CEB6B61/20240928193449

**Comment by zhuberty** [comment]
On Windows 11, I resolved this by ensuring that the Docker Desktop Service was actually running. My service was not set to start automatically for some reason.

In PowerShell (running as Administrator):
```powershell
Get-Service | Where-Object { $_.Name -like "*docker*" }

# result of the above command:
Status   Name               DisplayName
------   ----               -----------
Stopped  com.docker.service Docker Desktop Service
```

```powershell
Start-Service com.docker.service
```

```powershell
Get-Service | Where-Object { $_.Name -like "*docker*" }

# result of the above command:
Status   Name               DisplayName
------   ----               -----------
Running  com.docker.service Docker Desktop Service
```

To ensure that this service is set to start automatically:
```powershell
Set-Service -Name com.docker.service -StartupType Automatic
```

Verify that the service is set to start automatically:
```powershell
Get-Service com.docker.service | Select-Object Name, StartType

# result of the above command:
Name               StartType
----               ---------
com.docker.service Automatic
```

**Comment by lockedech** [comment]
> Solved by clearing these directories and reinstalling:
> 
> * `%APPDATA%\Docker`
> * `%USERPROFILE%\.docker`
> * `%USERPROFILE%\AppData\Local\Docker`

this is a bad answer, just like reinstalling the software. you may encounter this problem just because you have a VPN turned on.

**Comment by yizhixiaokong** [comment]
> > Solved by clearing these directories and reinstalling:
> > 
> > * `%APPDATA%\Docker`
> > * `%USERPROFILE%\.docker`
> > * `%USERPROFILE%\AppData\Local\Docker`
> 
> this is a bad answer, just like reinstalling the software. you may encounter this problem just because you have a VPN turned on.

You are correct!!! I reinstalled several times tried everything and still had the problem. After seeing your reply I closed the vpn and then he solved it! I tried several times over and over, the problem existed when I opened it and closed it and it was fine!!! wtf!

**Comment by 0xjairo** [comment]
in my case, I was using a Linux kernel we compiled internally with additional driver support for CAN. That is version `5.15.90.1-microsoft-standard-WSL2+`. Docker Desktop ran fine when I commented that out the `kernel=` line in `$env:USERPROFILE/.wslconfig`.

The default kernel in WSL (as of 2025-Jul-11) is `6.6.87.2-microsoft-standard-WSL2`

**Comment by realchrisolin** [comment]
Found this issue after encountering the same problem. While resetting the Docker directories under APPDATA and USERPROFILE got the engine to start again, it also blew out all my existing containers, images, and volumes.

Instead, I found the cause of why the engine wasn't starting logged in %USERPROFILE%\AppData\Local\Docker\log\vm\dockerd.log  (it was a config change). After reverting the breaking change, the engine started normally with all my existing data intact.
